### ISSUE TITLE
Support for Lian-Li TL controller?

### ISSUE THREAD

**Issue by MrDo-Arcade** [EightB1ts/FanControl.LianLi#18]
Just started looking in to Fan Control today, found that it has this plugin, which is cool... but then it hasn't been updated since June 2023, and in the original Reddit posts by the author announcing the plugin, the username is [deleted]... did something happen?

If still active... wondering if it's possible to add support for the new TL Controller that came out in December 2023?

Note that compared to previous LL controllers, this one shows up as two different devices in Device Manager:  looks like one is for the typical functions of the controller, and the second is specific to the LCD function on the TL LCD fans.  Also, while normal controllers only utilize one half of an internal USB header, this one utilizes the entire USB header (hence the two different devices).

Info from Device Manager 1:
- Name:  TL_Series_ControllerV0.1
- VID:  0416
- PID:  7372
- Complete string from Parent Device:  USB\VID_0416&PID_7372\Nuvoton
- Complete string from Hardware IDs 1:  USB\VID_0416&PID_7372&REV_0101&MI_00
- Complete string from Hardware IDs 2:  USB\VID_0416&PID_7372&MI_00

Info from Device Manager 2:
- Name:  TL_LCDV0.1
- VID: 04FC
- PID:  7393
- Complete string from Parent Device:  USB\VID_04FC&PID_7393\TL_LCDV0.1
- Complete string from Hardware IDs 1:  USB\VID_04FC&PID_7393&REV_0100&MI_01
- Complete string from Hardware IDs 2:  USB\VID_04FC&PID_7393&MI_01

If anything else is needed, please let me know.

**Comment by bonkas** [comment]
I am about to install these fans and the Uni Fan TL v2 controller in a week or two. I can also provide details if required..

+1 keen to know if this project is still alive.

**Comment by BthatKing** [comment]
I installed the Unifan TL Reverse with the Controller and would Love to use this Plugin.
Is there anyone working in it?

**Comment by Granville646** [comment]
Did anyone get the TL controller working?

**Comment by BthatKing** [comment]
Yes @Granville646
Enable Motherboard Passtrough in the LianLi li Software. 
Worked for me.

**Comment by Granville646** [comment]
> Yes @Granville646 Enable Motherboard Passtrough in the LianLi li Software. Worked for me.

I only see 1 sensor for all fans connected to the controller. Do you see all 4?

**Comment by BthatKing** [comment]
Did you Plug them in as one Block? Or did you Plug them in separately?

Its showing me one Fan too. But i built 3 Fans as one Block.

**Comment by Granville646** [comment]
> Did you Plug them in as one Block? Or did you Plug them in separately?
> 
> Its showing me one Fan too. But i built 3 Fans as one Block.

I have them in groups as i have a mix of TL, TL LCD and TL 140. They all operate at different RPM.

**Comment by bonkas** [comment]
> Yes @Granville646 Enable Motherboard Passtrough in the LianLi li Software. Worked for me.

I dont think this is exactly the solution. this will allow the fans to be controlled by the motherboard bios which kinda defeats the purpose? I think what myself and @Granville646 are looking for is a way for the fans to be directly software controlled, in my case it would be great to do without the LianLi software and control fan speeds via custom curves and custom temp sensors.

Also a side quest if anyone has a solution, getting support for these devices in linux...

**Comment by SeRoM** [comment]
+1 need to know if this project is still alive.

**Comment by Tymmey** [comment]
I'm trying myself on this, since i have TL, Tl-Reverse and TL-LCD-Reverse... However.
I can not find the fans in device manager. Only the controller itself :(

**Comment by Tymmey** [comment]
yeah without fan ids i'm getting nowhere...

**Comment by Enterprise12nx01** [comment]
im using the wireless versions plugged into my motherboard, any update on this?

**Comment by Tymmey** [comment]
dont expect anything on this.
The Code for the fans changed since the TL-Series.
We would need a complete recreation of this plugin....

Only way to make it working is:
Using one fancontroller for each fan row and set it to MB sync...
